diff --git a/appdaemon/dashboard.py b/appdaemon/dashboard.py
index eeb2a2a..836aa54 100755
--- a/appdaemon/dashboard.py
+++ b/appdaemon/dashboard.py
@@ -40,7 +40,7 @@ class Dashboard:
         self.images_dir = None
         self.base_url = ""
         self.dash_force_compile = False
-        self.dash_compile_on_start = False
+        self.dash_compile_on_start = True
         self.max_include_depth = 10
         self.fa4compatibility = False
         self.transport = "ws"
