--- a/appdaemon/plugins/hass/hassplugin.py
+++ b/appdaemon/plugins/hass/hassplugin.py
@@ -11,6 +11,7 @@ import datetime
 from urllib.parse import quote
 from urllib.parse import urlencode
 from typing import Union
+import os

 import appdaemon.utils as utils
 from appdaemon.appdaemon import AppDaemon
@@ -65,11 +66,17 @@ class HassPlugin(PluginBase):

         if "token" in args:
             self.token = args["token"]
+        elif "SUPERVISOR_TOKEN" in os.environ:
+            self.token = os.environ.get('SUPERVISOR_TOKEN')
+            self.AD.config["plugins"][name]["token"] = self.token
         else:
             self.token = None

         if "ha_url" in args:
             self.ha_url = args["ha_url"]
+        elif "SUPERVISOR_TOKEN" in os.environ:
+            self.ha_url = "http://supervisor/core"
+            self.AD.config["plugins"][name]["ha_url"] = self.ha_url
         else:
             self.logger.warning("ha_url not found in HASS configuration - module not initialized")
