--- a/appdaemon/plugins/hass/hassplugin.py
+++ b/appdaemon/plugins/hass/hassplugin.py
@@ -11,6 +11,7 @@ import datetime
 from urllib.parse import quote
 from urllib.parse import urlencode
 from typing import Union
+import os

 import appdaemon.utils as utils
 from appdaemon.appdaemon import AppDaemon
@@ -55,8 +56,8 @@ class HassPlugin(PluginBase):
         self.cert_path = args.get("cert_path")
         self.cert_verify = args.get("cert_verify")
         self.commtype = args.get("commtype", "WS")
-        self.ha_key = args.get("ha_key")
-        self.ha_url = args.get("ha_url", "")
+        self.ha_key = args.get("ha_key", os.environ.get("SUPERVISOR_TOKEN"))
+        self.ha_url = args.get("ha_url", "http://supervisor/core").rstrip("/")
         self.namespace = args.get("namespace", "default")
         self.plugin_startup_conditions = args.get("plugin_startup_conditions", {})
         self.retry_secs = int(args.get("retry_secs", 5))
